src/ray/gcs/gcs_server/gcs_server_main.cc

```diff
@@ -85,5 +92,5 @@ int main(int argc, char *argv[]) {
 
   gcs_server.Start();
 
-  main_service.run();
+  signal_service.run();
```

Comment:
Don't we need `boost::asio::io_service::work work` for this?

Reply:
It is definitely needed. I just renamed the variable so could not found it was missed in original codes.

Nice catch!